packages/myst-spec-ext/src/types.ts

```diff
@@ -36,6 +37,18 @@ export type CaptionNumber = Parent & {
   enumerator: string;
 };
 
+/**
+ * Line is, e.g., a line in an algorithm and can be numbered as well as indented.
+ * Otherwise this works the same as a paragraph, ideally with tighter styling.
+ * The Line is used in Algorithms (e.g. when parsing from LaTeX)
+ */
+export type Line = Parent & { type: 'line'; indent?: number; enumerator?: string };
```

Comment:
@fwkoch this is probably the most controversial change in the PR. I have added a new node of type `Line` which can have indentation and line numbers. This is to support the algorithm parsing, falling back to a div here is totally fine in the current theme.

Reply:
I did start there, adding attributes to paragraph, and went back and forth. I wanted the JAST to have a "specific-use" identifier and the UI to have a special style, and this seemed special enough for its own type. A div is also probably a more appropriate fallback based on style than a paragraph in this case?

At some point we should do an assessment of these types and do a clean up. I think that we can collapse a few things like `admonitionTitle` and `proof` --> `statement` perhaps.

I changed this to `algorithmLine` based on your first point!